x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -392,6 +389,16 @@ export class ESSearchSource extends AbstractESSource implements ITiledSingleLaye
     return !!(scalingType === SCALING_TYPES.TOP_HITS && topHitsSplitField);
   }
 
+  async loadIsEditable(): Promise<boolean> {
+    if (!getMapAppConfig().enableDrawingFeature) {
+      return false;
+    }
+    await this.getIndexPattern();
+    const { matchingIndexes } = await getMatchingIndexes(this.indexPattern!.title);
```

Comment:
"this.indexPattern!.title" may cause problems if the title is not provided. I think you should explicitly check that title is provided and return false of there is not title.